brian2/__init__.py

```diff
@@ -96,4 +99,51 @@ def _check_dependency_version(name, version):
     _check_dependency_version(name, version)
 
 # Initialize the logging system
-BrianLogger.initialize()
\ No newline at end of file
+BrianLogger.initialize()
+logger = get_logger(__name__)
+
+# Check the caches
+def _get_size_recursively(dirname):
+    total_size = 0
+    for dirpath, _, filenames in os.walk(dirname):
+        for fname in filenames:
+            total_size += os.path.getsize(os.path.join(dirpath, fname))
+    return total_size
+
+#: Stores the cache directory for code generation targets
+_cache_dirs = {}
+
+def check_cache(target):
+    cache_dir = _cache_dirs.get(target, None)
+    if cache_dir is None:
+        return
+    size = _get_size_recursively(cache_dir)
+    size_in_mb = int(round(size/1024./1024.))
+    if size_in_mb > prefs.codegen.max_cache_dir_size:
+        logger.info('Cache size for target "{target}": {size} MB.\n'
+                    'You can call "clear_cache(\'{target}\')" to delete all '
+                    'files from the cache or manually delete files in the '
+                    '"{cache_dir}" directory.'.format(target=target,
+                                                      size=size_in_mb,
+                                                      cache_dir=cache_dir))
+    else:
+        logger.debug('Cache size for target "%s": %s MB' % (target, size_in_mb))
+
+
+def clear_cache(target):
+    cache_dir = _cache_dirs.get(target, None)
+    if cache_dir is None:
+        raise ValueError('No cache directory registered for target "%s".' % target)
+    logger.debug('Clearing cache for target "%s" (directory "%s").' %
+                 (target, cache_dir))
+    shutil.rmtree(cache_dir)
```

Comment:
Are we absolutely 100% sure that ``cache_dir`` could never be something it shouldn't be?! :)

Reply:
Good point. Maybe we should ask the user for confirmation after displaying the name of the directory? Checking for file types sounds like a good idea as well.